### PR TITLE
alternate theme rendering - mass build pipeline

### PR DESCRIPTION
**Pull request by gumaerc**
#### Pre-Flight checklist

- [ ] Testing
  - [ ] Code is tested
  - [ ] Changes have been manually tested

#### What are the relevant tickets?
Closes https://github.com/mitodl/ocw-studio/issues/1428

#### What's this PR do?
This PR adds 3 optional arugments to the `upsert_mass_build_pipeline` management command:

 - `--prefix` - Adds an optional prefix to the path of sites deployed by the mass build pipeline
 - `--themes-branch` - Adds an optional override for the branch of `ocw-hugo-themes` to be used when building sites in the mass build pipeline
 - `--projects-branch` - Adds an optional override for the branch of `ocw-hugo-projects` to be used when building sites in the mass build pipeline

Each unique configuration will create a new set of pipelines for `draft` and `live` in the Concourse `mass-build-sites` group.  The only set of pipelines with automatic triggering enabled, however, will be the pipeline with the default branches of `ocw-hugo-themes` and `ocw-hugo-projects` matching `settings.GITHUB_WEBHOOK_BRANCH`.  Other mass build pipelines will need to be triggered manually.

#### How should this be manually tested?
 - Testing this PR will require setting up Concourse and Minio containers locally as well as a Github test organization, which only requires that you set some environment variables before running `docker-compose up`.  Check the readme for details.  You will also want to probably have a recent backup of the production database restored to your local instance by doing the following (replace `PROD_DB_URL` with the URL of the production database, `TODAY_DATE` with today's date):
```
pg_dump PROD_DB_URL > ./dumps/prod_TODAY_DATE.sql
docker-compose rm -sv db
docker-compose up -d db
docker-compose exec -T -e PGPASSWORD=postgres db psql -h db -U postgres < ./dumps/prod_TODAY_DATE.sql
docker-compose exec web ./manage.py createsuperuser
```
 -  Once your instance of `ocw-studio` is running, run the following to push up the theme assets pipeline and `ocw-www` pipeline:
```
docker-compose exec web ./manage.py upsert_theme_assets_pipeline
docker-compose exec web ./manage.py backpopulate_pipelines --filter ocw-www
```
- Once this is complete, browse to http://concourse:8080, login with test / test and go to the theme assets pipeline and trigger a build
- Once the theme assets build is complete, go to the live `ocw-www` pipeline and trigger a build
- Once both of these are complete, verify that you see the OCW home page at http://localhost:8045
- Make a temporary edit for testing to the `mass-build-sites.yml` template file (line 152) for this testing procedure to artifically limit the mass build to 10 sites:
```
            index=0
            jq -c '.sites[]' publishable_sites/sites.json | while read i; do
              if [[ $index < 10 ]]
              then
                result=$(process_site $i) || result=1
                echo "RESULT IS $result"
                if [[ $result == 1 ]]
                then
                  fail_site $i
                fi
                ((index=index+1))
              fi
            done
```
- Upsert the default mass build pipeline by simply running: `docker-compose exec web ./manage.py upsert_mass_build_pipeline`
- Upsert a test pipeline for building `course-v2` by running: `docker-compose exec web ./manage.py upsert_mass_build_pipeline --prefix /course-v2 --themes-branch main --projects-branch cg/course-v2-testing`
- Visit http://concourse:8080/?search=team%3A%22main%22%20group%3A%22mass-build-sites%22 and verify that you see 4 pipelines like this (yours will all be paused and grey):
image
- Unpause the `live` versions of the pipelines and start the one on the `main` branch (or whatever you have `GITHUB_WEBHOOK_BRANCH` set to)
- Once the pipeline completes, verify that you have a site at http://localhost:8045/courses/10-01-ethics-for-engineers-artificial-intelligence-spring-2020/ that is using the current version of the `course` theme:
image
- Run the mass build pipeline with `projects_branch: cg/course-v2-testing`
- Visit http://localhost:8045/course-v2/courses/10-01-ethics-for-engineers-artificial-intelligence-spring-2020/and verify that you have a site rendered with the `course-v2` theme like this:
image

#### Where should the reviewer start?
`upsert_mass_build_pipeline.py`, `concourse.py`, `concourse_test.py`, `mass-build-sites.yml`

#### Any background context you want to provide?
This will be used to build alternate versions of the site in RC for testing new themes

### CODE REVIEW

**Review by MAbdurrehman12 — APPROVED** [review]
Looks good 👍

File: content_sync/pipelines/definitions/concourse/mass-build-sites.yml [hunk 1]
```diff
@@ -155,6 +156,7 @@ jobs:
               then
                 fail_site $i
               fi
+            fi
```
Comment (mbertrand) [inline content_sync/pipelines/definitions/concourse/mass-build-sites.yml:159]: Getting this error:
```
selected worker: 12e736b474fb
++ pwd
+ CURDIR=/tmp/build/d1fb8792
+ [[ '' != '' ]]
+ GITKEYSSH=
bash: -c: line 54: syntax error near unexpected token `fi'
```

The pipeline completes successfully after removing the new `fi` line, but got an `Invalid BaseURL` error on every site:
```
Error: Error building site: logged 1 error(s)
++ return 1
+ result='Start building sites … 
hugo v0.98.0-165d299cde259c8b801abadc6d3405a229e449f6+extended linux/amd64 BuildDate=2022-04-28T10:23:30Z VendorInfo=gohugoio

ERROR 2022/08/02 15:09:11 Invalid BaseURL: "course-v2/courses/2-mb-site-2-spring-2025"
```

**Review by mbertrand — CHANGES_REQUESTED** [review]

**Review by mbertrand — APPROVED** [review]
👍

**Comment by pt2302** [comment]
@gumaerc Posting a reminder here to update the `concourse` container instructions for running on an M1 processor.

**Pull request merged**